### DISCUSSION TITLE
Driving a PMSM

### DISCUSSION THREAD

**Discussion by smistrie** [community/community#134259] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hello friends,

I have a small PMS motor here (Permanent Magnet Synchronous Motor, sometimes also called BLDC). I want to use this motor to run a current or torque control (Field-Oriented Control). I already have a driver for it. Now, I would like to make it spin. With a DC motor, you would simply connect the poles to a power source and the motor would spin. I want to achieve the same with the PMSM. My idea is to apply a sinusoidal voltage (in the form of PWM signals) to each of the three phases. The voltages are then shifted by 120° each. When I connect the driver, write my program, and turn on the power source, the motor briefly jumps, moves a little bit, and then stops. The power source indicates that no current is flowing (only 20mA). I have already checked all inputs and outputs on the driver with an oscilloscope. The PWM signals are definitely correct. I expected the motor to run. I suspect that at the moment of turning on the power source, a short, very high current flows, causing the driver to go into protection mode, and that's why nothing works. My idea would be to slowly increase the amplitude and frequency of the sinusoidal voltage to avoid high current spikes. But maybe it's something completely different. Can anyone help me? Is my approach even feasible?

Here are the parts used:

PMSM: GM2804 Gimbal Motor from iFlight (https://shop.iflight.com/ipower-gm2804-gimbal-motor-pro1153)
Driver: SimpleFOC Mini V1 (https://docs.simplefoc.com/simplefocmini)

Thank you very much!!!

**Comment by LiteBrite82** [comment]
Hi @smistrie ,

Thanks for being a part of the GitHub Community, we're glad you're here!

It's possible that another GitHub user might have run into this same issue and can help, but the GitHub Community Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

Best of luck!
